locals.tf

```diff
@@ -219,5 +211,32 @@ locals {
         }]
       }
     }
+
+    alloy = {
+      controller = {
+        type = "deployment"
+      }
+      alloy = {
+        serviceMonitor = {
+          enabled = true
```

Comment:
I think most of the bootstraps require us to deploy the Loki stack module before the kube-prometheus-stack module. As such, this could lead to issues, because the CRD for the ServiceMonitor would not exist. I propose you add a variable like in the modules cert-manager and Traefik that have a variable to enable the ServiceMonitor explicitly. This way it can be enabled after bootstrapping the cluster.